src/main/java/com/google/collegeplanner/servlets/CourseListServlet.java

```diff
@@ -31,21 +35,37 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: Store and retrieve from Datastore
-    String[] courseIdList = new String[] {"CMSC101", "CMSC106", "CMSC122", "CMSC131", "CMSC132",
-        "CMSC133", "CMSC216", "CMSC250", "CMSC298A", "CMSC320", "CMSC330", "CMSC351", "CMSC388J",
-        "CMSC389A", "CMSC389B", "CMSC389E", "CMSC389N", "CMSC390O"};
-
-    ArrayList<JSONObject> courses = new ArrayList<>();
-    for (String course : courseIdList) {
-      JSONObject newCourse = new JSONObject();
-      newCourse.put("name", course);
-      courses.add(newCourse);
+
+    // Specify the parameters for the API endpoint.
+    URIBuilder builder;
+    try {
+      builder = new URIBuilder("https://api.umd.io/v1/courses");
+      builder.setParameter("semester", "202008");
+    } catch (URISyntaxException e) {
+      respondWithError(
+          "Internal server error.", HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+      return;
+    }
+
+    JSONArray jsonArray = new APIUtil().requestAPIAndGetJsonArray(builder);
+    if (jsonArray == null) {
+      respondWithError(
+          "Internal server error.", HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
     }
 
     JSONObject schoolCourseInfo = new JSONObject();
-    schoolCourseInfo.put("courses_detailed", courses);
+    schoolCourseInfo.put("courses_detailed", jsonArray);
 
     response.setContentType("applications/json;");
     response.getWriter().println(schoolCourseInfo);
   }
+
+  private void respondWithError(String message, int errorType, HttpServletResponse response)
```

Comment:
No need to do it in this PR specifically, but I think we should create a `CollegePlannerServlet` which contains the `respondWithError` function, and then this class could be `class CourseListServlet extends CollegePlannerServlet`. That way, we can be consistent with our errors, and share other potential utils/functions/etc.

Please feel free to file an issue and follow up on this later.

Reply:
Filed an issue.